File: 백준/Gold/2230. 수 고르기/수 고르기.py
import sys
input = sys.stdin.readline

n, m = map(int, input().split())
nums = list(int(input()) for _ in range(n))
nums.sort()

start = 0
answer = sys.maxsize
for end in range(n):
    gap = nums[end] - nums[start]
    while m < gap:
        answer = min(gap, answer)
        start += 1
        gap = nums[end] - nums[start]
        if start >= end:
            break
    if gap == m:
        answer = m
        break

print(answer)

File: 백준/Gold/2230. 수 고르기/수 고르기.js
const input = require('fs').readFileSync(process.platform === 'linux' ? '/dev/stdin' : __dirname + '/input.txt').toString().trim().split('\n')

const [N, M] = input.shift().split(' ').map(Number)
const arr = input.map(Number)
arr.sort((a, b) => a - b)

const MAX_DIFF = 2_000_000_000
let answer = MAX_DIFF
let left = 0
let right = 0
while (left <= right && right < N) {
  const diff = Math.abs(arr[left] - arr[right])

  if (M <= diff) {
    answer = Math.min(answer, diff)
    left++
  } else {
    right++
  }
}
console.log(answer)